src/OrleansRuntime/Placement/StatelessWorkerDirector.cs

```diff
@@ -12,6 +12,26 @@ internal class StatelessWorkerDirector : IPlacementDirector<StatelessWorkerPlace
         public Task<PlacementResult> OnSelectActivation(
             PlacementStrategy strategy, GrainId target, IPlacementRuntime context)
         {
+            return Task.FromResult(SelectActivationCore(strategy, target, context));
+        }
+
+        public bool TrySelectActivationSynchronously(
+            PlacementStrategy strategy, GrainId target, IPlacementRuntime context, out PlacementResult placementResult)
+        {
+            placementResult = SelectActivationCore(strategy, target, context);
+            // stateless workers are always local, and  request for creation of a new activation will be created by PlacementDirectorsManager 
+            // in case of null returned by SelectActivationCore  
+            return true;
```

Comment:
if I am correct here and it should have indeed been return false on null, then I am very worried that we dont have any unit test about stateless workers that failed on this change. Can you please think about a test which would have failed in this case (if I am indeed correct about the bug). Without such a test, I don't feel comfortable abut this change.